drivers/fpga/fpga_ice40.c

```diff
@@ -207,6 +207,26 @@ static int fpga_ice40_load_gpio(const struct device *dev, uint32_t *image_ptr, u
 	struct fpga_ice40_data *data = dev->data;
 	const struct fpga_ice40_config *config = dev->config;
 
+	if (!device_is_ready(config->clk.port)) {
+		LOG_ERR("%s: GPIO for clk is not ready", dev->name);
+		return -ENODEV;
+	}
+
+	if (!device_is_ready(config->pico.port)) {
+		LOG_ERR("%s: GPIO for pico is not ready", dev->name);
+		return -ENODEV;
+	}
+
+	if (config->set == NULL) {
+		LOG_ERR("%s: set register was not specified", dev->name);
+		return -EFAULT;
+	}
+
+	if (config->clear == NULL) {
+		LOG_ERR("%s: clear register was not specified", dev->name);
+		return -EFAULT;
+	}
+
```

Comment:
It might be possible to move this check at build-time by removing the `_OR` from here:
https://github.com/zephyrproject-rtos/zephyr/blob/4bb945300bf92e4708404102d2a3b54e540bfd6c/drivers/fpga/fpga_ice40.c#L536

But it is still allowed to be empty for SPI, so it could be some build assert checking `FPGA_ICE40_LOAD_MODE(inst)` eventually.

A run-time check is good too!

Reply:
+1 for build-time checks.